### PR TITLE
Fix Short field

### PR DESCRIPTION
**Pull request by tiny1990**
if table schema is tinyint  not null, convert to  kafka schema is INT16.

### CODE REVIEW

**Comment by jpechane** [comment]
Looks good, thanks! Could you please extract the constant into a new class in `util` package in the similar vein like https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/math/NumberUtils.html ?

**Comment by tiny1990** [comment]
@jpechane  This PR is only fix Short convert bug?

**Comment by jpechane** [comment]
Well as you told, it is an constnt so maybe it is better to handle it as a constant instead of creating a new class instance for every event coming. Thus I belive it belongs to this PR.

**Comment by tiny1990** [comment]
@jpechane    review please

**Comment by gunnarmorling** [comment]
Hi @tiny1990, thanks for contributing! Looking into the PR, where is the actual change, apart from the refactoring?

**Comment by jpechane** [comment]
It is line 794. Originally change was `new Short((short)0)` and I've asked for using a predefined constant extracted into a separate file. @tiny1990 has extracted all the constants. I am sorry for the confusion.

**Comment by tiny1990** [comment]
@gunnarmorling https://github.com/debezium/debezium/blob/master/debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java#L804 should return Short object,  0 is Integer , Kafka schema validate failed

File: debezium-core/src/main/java/io/debezium/util/NumberConstants.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.util.Date;
+
+/**
+ * A set of Numbers.
+ *
+ * @author pmx
+ */
+public class NumberConstants {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/util/NumberConstants.java]: Could you add your real name? Makes it easier to grasp the history and work with this code down the road :)

File: debezium-core/src/main/java/io/debezium/util/NumberConstants.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    public static final Double DOUBLE_FALSE = new Double(0.0d);
+
+    public static final byte[] BYTE_ZERO = new byte[0];
+
+    public static final Date DATE_EPOCH_ZERO = new Date(0L);
+
+    public static final BigDecimal BIGDECIMAL_ZERO = BigDecimal.valueOf(0);
+    public static final BigDecimal BIGDECIMAL_ONE = BigDecimal.valueOf(1);
+
+    public static final ByteBuffer BYTEBUFFER_ZERO = ByteBuffer.wrap(new byte[0]);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/util/NumberConstants.java]: Is that safe for concurrent use? Although it's empty, I don't have a good feeling about sharing `ByteBuffer`.

File: debezium-core/src/main/java/io/debezium/util/NumberConstants.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Convert boolean object to bigDecimal object.
+     *
+     * @param data a boolean object
+     * @return bigDecimal 0 or 1
+     * @throws NullPointerException If {@code data} is {@code null}
+     */
+    public static BigDecimal getBigDecimal(Boolean data) {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/util/NumberConstants.java]: Given we have those methods here, I think the class shouldn't be named `...Constants` (I'd keep that for interfaces solely containing constants). How about something like `NumberConversions`?

File: debezium-core/src/main/java/io/debezium/util/NumberConstants.java (hunk 2 above)
Comment (tiny1990) [reply debezium-core/src/main/java/io/debezium/util/NumberConstants.java]: yes , wrap a new object rather than use a static object .

**Comment by tiny1990** [comment]
@gunnarmorling  
[√]1. file in https://issues.jboss.org/projects/DBZ/issues/DBZ-422?filter=allopenissues
[√]2. split 2 commit 
[√]3. add  test
review PR plz.

**Comment by gunnarmorling** [comment]
@tiny1990, LGTM overall. I'll just change the commit and apply some clean-ups here and there and then merge.

The one thing I'm wondering is this: how do you run into this situation actually? If the column is `NON NULL`, in which case does it happen that `null` is received in `convertSmallInt()`. The fix is good either way, I'm just curios, when this actually happens.

**Comment by gunnarmorling** [comment]
Applied to master and 0.6 and added one more commit with the clean-up discussed above. Thanks, @tiny1990!

**Comment by tiny1990** [comment]
@gunnarmorling 
When the table is empty  I'll send "Zero record" to kafka.
Notify select mode or binlog mode to sink connector.

**Comment by gunnarmorling** [comment]
Hey @tiny1990, could you give me your name so we can add it to the list of contributors. Thanks!

**Comment by tiny1990** [comment]
Pan Maoxiang @gunnarmorling

**Comment by gunnarmorling** [comment]
Excellent, thank you very much!

**Comment by tiny1990** [comment]
my pleasure. But I think use "Maoxiang Pan" is best, because I'm Chinese.

**Comment by gunnarmorling** [comment]
Ok, I'll add "Maoxiang Pan" then. The different order of names always tripped me up :)

**Pull request closed without merging**